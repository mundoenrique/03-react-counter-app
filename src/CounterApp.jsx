import PropTypes from 'prop-types';
import { useState } from 'react';
export default function CounterApp({ value }) {
	const [counter, setCounter] = useState(value);

	const handleAdd = () => {
		setCounter(counter + 1);
	};

	const handleReset = () => setCounter(value);
	const handleSubstract = () => setCounter((c) => c - 1);

	return (
		<>
			<h1>Counter App</h1>
			<h2>{counter}</h2>
			<button onClick={handleAdd}>+1</button>
			<button aria-label="btn-reset" onClick={handleReset}>
				reset
			</button>
			<button data-testid="test-1" onClick={handleSubstract}>
				-1
			</button>
		</>
	);
}

CounterApp.prototype = {
	value: PropTypes.number.isRequired,
};
