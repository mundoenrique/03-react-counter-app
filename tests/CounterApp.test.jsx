import { fireEvent, render, screen } from '@testing-library/react';
import CounterApp from '../src/CounterApp';

describe('Pruebas en <CounterApp />', () => {
	const value = 10;

	test('Debe hacer natch con el snapshot', () => {
		const { container } = render(<CounterApp value={value} />);

		expect(container).toMatchSnapshot();
	});

	test('Debe mostrar el valor inicial de 100', () => {
		render(<CounterApp value={100} />);

		expect(screen.getByText('100')).toBeTruthy();
		expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain(
			'100'
		);
	});

	test('Debe incrementar en un con el botón +1', () => {
		render(<CounterApp value={value} />);
		fireEvent.click(screen.getByText('+1'));

		expect(screen.getByText(value + 1)).toBeTruthy();
	});

	test('Debe incrementar en un con el botón -1', () => {
		render(<CounterApp value={value} />);
		fireEvent.click(screen.getByText('-1'));

		expect(screen.getByText(value - 1)).toBeTruthy();
	});

	test('Debe incrementar funcionar el botón reset', () => {
		render(<CounterApp value={value} />);

		fireEvent.click(screen.getByText('-1'));
		fireEvent.click(screen.getByText('-1'));
		fireEvent.click(screen.getByText('-1'));
		fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));

		expect(screen.getByText(value)).toBeTruthy();
	});
});
