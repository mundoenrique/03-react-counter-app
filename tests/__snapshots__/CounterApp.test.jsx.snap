// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debe hacer natch con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    10
  </h2>
  <button>
    +1
  </button>
  <button
    aria-label="btn-reset"
  >
    reset
  </button>
  <button
    data-testid="test-1"
  >
    -1
  </button>
</div>
`;
